From 6729ade2742cae161196fe3a6f5168e40a075b7c Mon Sep 17 00:00:00 2001
From: Alice <alicem@gnome.org>
Date: Fri, 12 Sep 2025 23:33:59 +0400
Subject: [PATCH 10/18] toast-widget: Make the button insensitive after a click

Otherwise we risk emitting ::button-clicked multiple times.

Fixes https://gitlab.gnome.org/GNOME/libadwaita/-/issues/1070


(cherry picked from commit fe38eefd9b8c14efbf8758c0e593942ff3cd3d05)

Co-authored-by: Alice Mikhaylenko <alicem@gnome.org>

(cherry picked from commit 15f32c4795f02c58a45b900f9437898060bc81ae)

Co-authored-by: Alice Mikhaylenko <alicem@gnome.org>
---
 src/adw-toast-widget.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/adw-toast-widget.c b/src/adw-toast-widget.c
index 77dd167e..10ba8323 100644
--- a/src/adw-toast-widget.c
+++ b/src/adw-toast-widget.c
@@ -102,6 +102,8 @@ action_clicked_cb (AdwToastWidget *self)
 {
   end_timeout (self);
 
+  gtk_widget_set_sensitive (self->action_button, FALSE);
+
   g_signal_emit_by_name (self->toast, "button-clicked");
 
   /* Keep the widget alive through the idle. Otherwise it may be immediately
-- 
2.51.1

